policy_definitions/domainjoin_windows/modules/archetypes/lib/policy_assignments/policy_definition_qby_deploy_windows_domainjoin.json

```
@@ -0,0 +1,273 @@
+{
+  "name": "Deploy-Windows-DomainJoin",
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "apiVersion": "2021-06-01",
+  "scope": null,
+  "properties": {
+    "policyType": "Custom",
+    "mode": "Indexed",
+    "displayName": "Deploy Windows Domain Join Extension with keyvault configuration",
+    "description": "Deploy Windows Domain Join Extension with keyvault configuration when the extension does not exist on a given windows Virtual Machine",
+    "metadata": {
+      "version": "1.0.0",
+      "category": "Guest Configuration",
+      "source": "https://github.com/Azure/Enterprise-Scale/",
+      "alzCloudEnvironments": [
+        "AzureCloud",
+        "AzureChinaCloud",
+        "AzureUSGovernment"
+      ]
+    },
+    "parameters": {
+      "domainUsername": {
+        "type": "String",
+        "metadata": {
+          "displayName": "domainUsername"
+        }
```

```suggestion
        "metadata": {
          "displayName": "SecretNameUsername"
          "description": "Name of the secret holding the username to use for domain join."
        }
```

You could also use `"strongType":"Microsoft.KeyVault/vaults/secrets"` to make sure only secrets are added. But that would require changes in the policy.